### ISSUE TITLE
How to readJSON for completely empty input? 

### ISSUE THREAD

**Issue by cah6** [justinwoo/purescript-simple-json#66]
Hey there,

I'm using Affjax to execute requests, defining the response format as String, and then using readJSON to parse the response into some type defined by the caller. I have a request, though, where the success response is nothing, i.e. completely empty. I tried defining the `a` in readJSON to be String, Maybe String, Nullable String, Foreign, etc but it all returns a parse error on the readJSON call. What I think I need is an instance for ReadForeign Unit that looks like:
```
instance readForeignUnit :: ReadForeign Unit where
  readImpl = pure unit
```
such that callers that expect the response to be completely empty can define that in the type and have a useless result be returned back in that slot. Can you foresee any issues with this or do you have any other suggestions for what to do here?

**Comment by justinwoo** [comment]
`Unit` is not meant to have some JSON representation or anything. It seems like what you should be doing is defining a different function to call upstream such that the response bodies aren't automatically parsed.

Regardless what you use, the initial JSON parsing will fail when dealing with the invalid JSON string. ReadForeign really only deals with how to deal with actual `Foreign` values being decoded, not the actual parsing of the JSON string.

**Comment by justinwoo** [comment]
You don't have to take my word for it, look at how the readJSON methods work by first calling parseJSON, which is a separate function: https://github.com/justinwoo/purescript-simple-json/blob/c5e2b7b25e2e30fb9940cad388377a2040d8b54a/src/Simple/JSON.purs#L79

**Comment by cah6** [comment]
Ack, sorry for the delay here. You're right, of course; when I made this ticket I had forgotten that a completely blank input is indeed invalid JSON.

For anybody else that might stumble on this: what we ended up doing was parameterizing our request method on a "how to handle the response" input (more or less of type `String -> E a`) and giving it `readJSON` in the usual case but giving it `const (Right unit)` in the case where the expected response is unit.

Closing this, thank you!
